scripts/o3de/o3de/project_manager_interface.py

```diff
@@ -138,7 +139,39 @@ def get_all_project_infos() -> list:
 
         :return list of dicts containing project infos.
     """
-    return list()
+    project_paths = manifest.get_all_projects()
+
+    # get all engine info once up front
+    engines_json_data = OrderedDict()
+    engines = manifest.get_manifest_engines()
+    for engine in engines:
+        if isinstance(engine, dict):
+            engine_path = pathlib.Path(engine['path']).resolve()
+        else:
+            engine_path = pathlib.Path(engine).resolve()
+        engine_json_data = manifest.get_engine_json_data(engine_path=engine_path)
+        if not engine_json_data:
+            continue
+        engines_json_data[engine_path] = engine_json_data
```

Comment:
This logic seems like it should be in a method on its own.
i.e `refresh_engine_info_cache`

Reply:
I like it, something similar to `get_gems_json_data_by_name`